v2/tools/generator/root.go

```diff
@@ -17,7 +21,17 @@ func Execute() {
 		klog.Fatalf("fatal error: commands failed to build! %s\n", err)
 	}
 
-	if err := cmd.Execute(); err != nil {
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
Any reason not to put this bit into a shared function?

`makeSignalCtx` or something?

Reply:
Good idea. I just need to work out where that shared function should live.

Reply:
Done.